src/factory/ZoraCreator1155FactoryImpl.sol

```diff
@@ -95,5 +95,9 @@ contract ZoraCreator1155FactoryImpl is IZoraCreator1155Factory, ContractVersionB
     /// @notice Ensures the caller is authorized to upgrade the contract
     /// @dev This function is called in `upgradeTo` & `upgradeToAndCall`
     /// @param _newImpl The new implementation address
-    function _authorizeUpgrade(address _newImpl) internal override onlyOwner {}
+    function _authorizeUpgrade(address _newImpl) internal override onlyOwner {
+        if (keccak256(abi.encode(IContractMetadata(_newImpl).contractName())) != keccak256(abi.encode(this.contractName()))) {
```

Comment:
Would this be cleaner as a library call?

`StringUtils.eq(this.contractName(), IContractMetadata(_newImpl).contractName())`?

Reply:
feel like its not worth bringing in stringutils for just this

Reply:
Just that it's hard to read as-is for me as to what this is doing.

We already have Strings. used elsewhere, perhaps use that with `Strings.equal` or their impl which is:
```

    /**
     * @dev Returns true if the two strings are equal.
     */
    function equal(string memory a, string memory b) internal pure returns (bool) {
        return keccak256(bytes(a)) == keccak256(bytes(b));
    }
```